Providers/PythonProvider.cpp

```diff
@@ -655,6 +633,10 @@ PythonProvider::verifySocketState ()
     }
     if (INVALID_SOCKET == m_FD)
     {
+        if(m_pid > 0 )
+        {
+           m_PreviousPid.push_back(m_pid);
+        }
         result = init ();
```

Comment:
why put in vector instead of doing waitpid here ? the above does not look like blocking call as process probably already is zombie ?